ios/HybridImageFactory.swift

```diff
@@ -21,14 +22,102 @@ class HybridImageFactory: HybridImageFactorySpec {
     guard let url = URL(string: urlString) else {
       throw RuntimeError.error(withMessage: "URL string \"\(urlString)\" is not a valid URL!")
     }
-
+    
     return Promise.async {
       let webImageOptions = options?.toSDWebImageOptions() ?? []
       let uiImage = try await SDWebImageManager.shared.loadImage(with: url, options: webImageOptions)
       return HybridImage(uiImage: uiImage)
     }
   }
   
+  /**
+   * Load Image from URL
+   */
+  func loadFromAssetAsync(assetId: String, options: AssetImageLoadOptions?) throws -> Promise<any HybridImageSpec> {
+    let assets = PHAsset.fetchAssets(
+      withLocalIdentifiers: [assetId],
+      options: nil
+    )
+    guard let asset = assets.firstObject else {
+      throw NSError(
+        domain: "TurboImageView",
+        code: 404,
+        userInfo: [NSLocalizedDescriptionKey: "Asset not found"]
+      )
+    }
+    
+    return Promise.async {
+      return try await withCheckedThrowingContinuation { continuation in
```

Comment:
Don't write `withCheckedThrowingContinuation` inline. Create an extension on `PHImageManager` that makes this thing async instead of trying to force everything into a single method.

Add a new file, call it `PHImageManager+requestImage` or something, add the `extension` on `PHImageManager` and add a single (or two) asynchronous methods that perform the stuff you are doing here.

Also, use `Checked` only for debugging, and `Unchecked` for release. No need to check if we call the result each time.